enterprise/frontend/src/embedding-sdk/components/private/IsolatedStoreProvider.tsx

```diff
@@ -0,0 +1,50 @@
+import { type ComponentType, type ReactNode, type FC, useMemo } from "react";
+import { Provider } from "react-redux";
+
+import { sdkReducers, useSdkSelector } from "embedding-sdk/store";
+import { getLoginStatus } from "embedding-sdk/store/selectors";
+import { useStore } from "metabase/lib/redux";
+import { getStore } from "metabase/store";
+
+interface Props {
+  children: ReactNode;
+}
+
+export const IsolatedStoreProvider = ({ children }: Props) => {
+  const store = useStore();
+  const loginStatus = useSdkSelector(getLoginStatus);
+
+  // Capture the root store's initial state when the login status is successful.
+  const isolatedStore = useMemo(
+    () => {
+      if (loginStatus.status !== "success") {
+        return null;
+      }
+
+      return getStore(sdkReducers, null, store.getState());
+    }, // eslint-disable-next-line react-hooks/exhaustive-deps
+    [loginStatus],
```

Comment:
I tried to use `useState` with `useEffect` instead of `useMemo` so I can avoid the exhaustive-deps error, but this causes all unit tests to fail even though it also works on the browser. The reason we cannot add "store" to the dep array is that it would reconstruct the store every time.

I also tried importing `store` directly from `metabase/store` but that caused the test to fail too, so only this exact setup with `useMemo` and `useStore` work with unit tests so far. It's strange.